tests/unit/test_pipeline.py

```diff
@@ -53,12 +59,39 @@ def test_get_diagnostic_dir(default_config, mocker):
     )
 
 
-def test_build_snakefile(default_config, mocker, test_dir):
+def test_build_snakefile_local(default_config, mocker, test_dir):
     mocker.patch("linker.implementation.Implementation.validate", return_value={})
     pipeline = Pipeline(default_config)
     with TemporaryDirectory() as snake_dir:
         snakefile = pipeline.build_snakefile(Path(snake_dir))
-        expected_file_path = Path(os.path.dirname(__file__)) / "rule_strings/pipeline.txt"
+        expected_file_path = (
+            Path(os.path.dirname(__file__)) / PIPELINE_STRINGS["pipeline_local"]
+        )
+        with open(expected_file_path) as expected_file:
+            expected = expected_file.read()
+        expected = expected.replace("{snake_dir}", snake_dir)
+        expected = expected.replace("{test_dir}", test_dir)
+        snake_str = snakefile.read_text()
+        snake_str_lines = snake_str.split("\n")
+        expected_lines = expected.split("\n")
+    assert len(snake_str_lines) == len(expected_lines)
+    for i, expected_line in enumerate(expected_lines):
+        assert snake_str_lines[i].strip() == expected_line.strip()
+
+
+def test_build_snakefile_slurm(default_config_params, mocker, test_dir):
```

Comment:
I didn't look super close but these two tests seem to be lots of duplicate code - can you parameterize?

Reply:
done